src/StyledOcticon/StyledOcticon.tsx

```diff
@@ -1,18 +1,5 @@
-import {IconProps} from '@primer/octicons-react'
-import React from 'react'
-import styled from 'styled-components'
-import sx, {SxProp} from '../sx'
-import {ComponentProps} from '../utils/types'
+// Leaving an export here for backwards compatibility
+import {default as StyledOcticon} from '../Octicon/Octicon'
 
-type OcticonProps = {icon: React.ElementType; color?: string} & IconProps
-
-function Octicon({icon: IconComponent, ...rest}: OcticonProps) {
-  return <IconComponent {...rest} />
-}
-
-const StyledOcticon = styled(Octicon)<SxProp>`
-  ${({color, sx: sxProp}) => sx({sx: {color, ...sxProp}})}
-`
-
-export type StyledOcticonProps = ComponentProps<typeof StyledOcticon>
 export default StyledOcticon
```

Comment:
Can we add a deprecation tag to StyledOcticon?

```suggestion
/** @deprecated Use Octicon instead */
export default StyledOcticon
```